src/test/java/org/springframework/integration/aws/support/S3SessionTests.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aws.support;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+/**
+ * @author Xavier François
+ */
+class S3SessionTests {
+
+	@Test
+	void getHostPortWithoutEndpoint() {
+
+		AmazonS3 amazonS3 = AmazonS3ClientBuilder.standard().withRegion(Regions.EU_WEST_1).build();
```

Comment:
I don't think we should use a real Amazon client. Not everyone running this project is going to have AWS account.

Please, consider to adjust `S3InboundChannelAdapterTests` and/or `S3StreamingChannelAdapterTests` which are based on the mock for `AmazonS3`.

I mean we indeed don't need this extra test class for such a simple functionality to verify.

Reply:
It's indeed a simple functionality, I strongly agree with you on that point. But I struggle to gracefully integrate the tests to existing ones. There is just one test in each of `S3InboundChannelAdapterTests`  and `S3StreamingChannelAdapterTests`  classes and I think my tests won't integrate in them really well.  What I can do is moving moving my `getHostPortWithEndpoint` test to `S3InboundChannelAdapterTests` using the provided mock in the nested Config. But I'm not sure it's what you are expecting.